diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 3218a733b50c..d5a9329ed28d 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -1504,7 +1504,7 @@ pref("browser.tabs.remote.desktopbehavior", true);
 // We leave it here set to false to reset users' defaults and allow
 // us to change everybody to true in the future, when desired.
 pref("browser.tabs.remote.autostart.1", false);
-pref("browser.tabs.remote.autostart.2", true);
+pref("browser.tabs.remote.autostart.2", false;
 #endif
 
 // For the about:tabcrashed page
