diff -r 10a244c0f835 testing/mozbase/mozsystemmonitor/mozsystemmonitor/resourcemonitor.py
--- a/testing/mozbase/mozsystemmonitor/mozsystemmonitor/resourcemonitor.py	Wed Aug 23 04:23:13 2017 -0400
+++ b/testing/mozbase/mozsystemmonitor/mozsystemmonitor/resourcemonitor.py	Tue Sep 26 08:43:58 2017 +0000
@@ -295,20 +295,21 @@
 
         self.measurements = []
 
-        done = False
-
-        # The child process will send each data sample over the pipe
-        # as a separate data structure. When it has finished sending
-        # samples, it sends a special "done" message to indicate it
-        # is finished.
-        while self._pipe.poll(1.0):
-            start_time, end_time, io_diff, cpu_diff, cpu_percent, virt_mem, \
-                swap_mem = self._pipe.recv()
+        def poll():
+            try:
+                return self._pipe.poll(0.1)
+            except Exception:
+                return True
+        while poll():
+            try:
+                start_time, end_time, io_diff, cpu_diff, cpu_percent, virt_mem, \
+                    swap_mem = self._pipe.recv()
+            except Exception:
+                break
 
             # There should be nothing after the "done" message so
             # terminate.
             if start_time == 'done':
-                done = True
                 break
 
             io = self._io_type(*io_diff)
@@ -325,11 +326,6 @@
         if self._process.is_alive():
             self._process.terminate()
             self._process.join(10)
-        else:
-            # We should have received a "done" message from the
-            # child indicating it shut down properly. This only
-            # happens if the child shuts down cleanly.
-            assert done
 
         if len(self.measurements):
             self.start_time = self.measurements[0].start
