error: expected identifier, found keyword `self`
  --> $DIR/ref-self.rs:10:23
   |
10 |     fn method_ref(ref self: Pin<&mut Self>) {} //~ ERROR expected identifier, found keyword `self`
   |                       ^^^^ expected identifier, found keyword

error: expected identifier, found keyword `self`
  --> $DIR/ref-self.rs:11:31
   |
11 |     fn method_ref_mut(ref mut self: Pin<&mut Self>) {} //~ ERROR expected identifier, found keyword `self`
   |                               ^^^^ expected identifier, found keyword
