error[E0277]: `std::marker::PhantomPinned` cannot be unpinned
  --> $DIR/add-pinned-field.rs:21:5
   |
4  | fn is_unpin<T: Unpin>() {}
   |                ----- required by this bound in `is_unpin`
...
21 |     is_unpin::<Foo>(); //~ ERROR E0277
   |     ^^^^^^^^^^^^^^^ within `__Foo<'_>`, the trait `std::marker::Unpin` is not implemented for `std::marker::PhantomPinned`
   |
   = note: required because it appears within the type `__Foo<'_>`
   = note: required because of the requirements on the impl of `std::marker::Unpin` for `Foo`

error[E0277]: `std::marker::PhantomPinned` cannot be unpinned
  --> $DIR/add-pinned-field.rs:22:5
   |
4  | fn is_unpin<T: Unpin>() {}
   |                ----- required by this bound in `is_unpin`
...
22 |     is_unpin::<Bar>(); //~ ERROR E0277
   |     ^^^^^^^^^^^^^^^ within `__Bar<'_>`, the trait `std::marker::Unpin` is not implemented for `std::marker::PhantomPinned`
   |
   = note: required because it appears within the type `__Bar<'_>`
   = note: required because of the requirements on the impl of `std::marker::Unpin` for `Bar`
