error[E0277]: `std::marker::PhantomPinned` cannot be unpinned
 --> $DIR/trivial_bounds-feature-gate.rs:8:5
  |
8 |     impl Unpin for A where PhantomPinned: Unpin {} //~ ERROR E0277
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::marker::Unpin` is not implemented for `std::marker::PhantomPinned`
  |
  = help: see issue #48214
  = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: `std::marker::PhantomPinned` cannot be unpinned
   --> $DIR/trivial_bounds-feature-gate.rs:8:43
    |
8   |     impl Unpin for A where PhantomPinned: Unpin {} //~ ERROR E0277
    |                                           ^^^^^ the trait `std::marker::Unpin` is not implemented for `std::marker::PhantomPinned`

error[E0277]: `std::marker::PhantomPinned` cannot be unpinned
  --> $DIR/trivial_bounds-feature-gate.rs:16:5
   |
16 |     impl Unpin for B where Wrapper<PhantomPinned>: Unpin {} //~ ERROR E0277
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::marker::Unpin` is not implemented for `std::marker::PhantomPinned`
   |
   = note: required because of the requirements on the impl of `std::marker::Unpin` for `phantom_pinned::Wrapper<std::marker::PhantomPinned>`
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: `std::marker::PhantomPinned` cannot be unpinned
  --> $DIR/trivial_bounds-feature-gate.rs:33:5
   |
33 |     impl Unpin for A where Inner: Unpin {} //~ ERROR E0277
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ within `inner::Inner`, the trait `std::marker::Unpin` is not implemented for `std::marker::PhantomPinned`
   |
   = note: required because it appears within the type `inner::Inner`
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: `std::marker::PhantomPinned` cannot be unpinned
  --> $DIR/trivial_bounds-feature-gate.rs:41:5
   |
41 |     impl Unpin for B where Wrapper<Inner>: Unpin {} //~ ERROR E0277
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ within `inner::Inner`, the trait `std::marker::Unpin` is not implemented for `std::marker::PhantomPinned`
   |
   = note: required because it appears within the type `inner::Inner`
   = note: required because of the requirements on the impl of `std::marker::Unpin` for `inner::Wrapper<inner::Inner>`
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable
